can/bus.py

```diff
@@ -417,17 +419,34 @@ def flush_tx_buffer(self) -> None:
 
     def shutdown(self) -> None:
         """
-        Called to carry out any interface specific cleanup required
-        in shutting down a bus.
+        Called to carry out any interface specific cleanup required in shutting down a bus.
+
+        This method can be safely called multiple times.
         """
+        if self._is_shutdown:
+            LOG.debug("%s is already shut down", self.__class__)
+            return
+
         self.stop_all_periodic_tasks()
+        self._is_shutdown = True
 
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.shutdown()
 
+    def __del__(self) -> None:
+        try:
+            if not self._is_shutdown:
+                # We do some best-effort cleanup if the user
+                # forgot to properly close the bus instance
+                self.shutdown()
+                LOG.warning("%s was not properly shut down", self.__class__)
+        except Exception as e:  # pylint: disable=W0703
+            # Prevent unwanted output from being printed to stdout/stderr
+            LOG.debug(e)
+
```

Comment:
Why would we want to supress exceptions there? Seems like an excpetion printed like [here](https://github.com/redis/redis-py/issues/1339#issue-614826236) would be quite useful, right?

Also a small suggestion as follows. That way, the user gets the information about the shutdown as context before the stack trace.
```suggestion
        if not self._is_shutdown:
            LOG.warning("%s was not properly shut down", self.__class__)
            # We do some best-effort cleanup if the user
            # forgot to properly close the bus instance
            try:
                self.shutdown()
            except Exception as e:  # pylint: disable=W0703
                # Prevent unwanted output from being printed to stdout/stderr
                LOG.debug(e)
```

Reply:
Hmm I think you're right. My thought was that because `__del__` _always_ gets called that it could end up adding confusion to the stack trace. 

But after testing I see that as long as `_is_shutdown` is true it won't result in double output. I need to swap lines 434 & 435 to ensure this always happens.

Reply:
Also without this try/except block the `tox` test report gets littered with captured warnings from existing ut's. I opted to set `pytest.filterwarnings` to ignore these.